colossalai/auto_parallel/solver/sharding_strategy.py

```diff
@@ -69,6 +90,20 @@ class TrainCycleItem:
     total: Any
 
 
+class CommunicationType(Enum):
```

Comment:
I think we could unify all the communication related operation to CommSpec, which could give communication cost information using get_comm_cost api, and support runtime application with convert_spec_to_action. The resharding costs is estimated by ShapeConsistencyManager which uses CommSpec internally.